npm/src/graphql/index.ts

```diff
@@ -0,0 +1,81 @@
+import { createHttpLink } from "apollo-link-http"
+import {
+  ApolloClient,
+  InMemoryCache,
+  ApolloQueryResult,
+  DocumentNode,
+  NormalizedCacheObject,
+  QueryOptions,
+  MutationOptions,
+  FetchResult
+} from "apollo-boost"
+import { getToken } from "../auth"
+import fetch from "unfetch"
+
+type CommonQueryOptions<T> = Omit<T, "query">
+
+export class GraphqlClient {
+  client: ApolloClient<NormalizedCacheObject>
+
+  constructor(uri: string) {
+    const link = createHttpLink({
+      uri,
+      fetch
+    })
+
+    this.client = new ApolloClient({
+      link,
+      cache: new InMemoryCache()
+    })
+  }
+
+  private getOptions: <D, V>(
+    variables: V
+  ) => Promise<
+    CommonQueryOptions<QueryOptions<V> | MutationOptions<D, V>>
+  > = async <V>(variables: V) => {
+    try {
+      const token = await getToken()
+      return {
+        variables,
+        context: {
+          headers: {
+            Authorization: `Bearer ${token}`
+          }
+        }
+      }
+    } catch {
+      throw Error("Cannot fetch token")
```

Comment:
I've always been a bit hazy on good practices for handling errors in JavaScript, but can we include the details of the original error rather than throwing them away, so it's easier to debug?

Reply:
Good point. I've thrown the original error and updated the tests.